File: src/lec04/Lec04Main.java
package lec04;

public class Lec04Main {
    public static void main(String[] args) {
        JavaMoney money1 = new JavaMoney(1_000L);
        JavaMoney money2 = new JavaMoney(2_000L);
        System.out.println(money1.plus(money2));

        JavaMoney money3 = new JavaMoney(1_000L);
        JavaMoney money4 = money3;
        JavaMoney money5 = new JavaMoney(1_000L);

        System.out.println(money3 == money4);   // true
        System.out.println(money3 == money5);   // false
        System.out.println(money3.equals(money5));   // true

        JavaMoney money6 = new JavaMoney(1_000L);
        JavaMoney money7 = new JavaMoney(2_000L);
        System.out.println(money6.plus(money7));
    }
}

File: src/lec04/Lec04Main.kt
package lec04

fun main() {
    val money1 = JavaMoney(2_000L)
    val money2 = JavaMoney(1_000L)

    if (money1 > money2) {  // 객체를 비교할 때 compareTo를 자동으로 호출한다.
        println("Money1이 Money2보다 금액이 큽니다")
    }

    val money3 = JavaMoney(1_000L)
    val money4 = money3
    val money5 = JavaMoney(1_000L)

    println(money3 === money4)  // true
    println(money3 === money5)  // false
    println(money3 == money5)  // true

    val money6 = Money(1_000L)
    val money7 = Money(2_000L)
    println(money6 + money7)    // Money(amount=3000)
}
